### PR TITLE
Docs: Fix identifiers and links of branching

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by manuzhang** [comment]
@nastra please help review. Thanks!

File: docs/flink-writes.md [hunk 1]
```diff
@@ -193,7 +193,7 @@ FlinkSink.builderFor(
 
 ### Branch Writes
 Writing to branches in Iceberg tables is also supported via the `toBranch` API in `FlinkSink`
-For more information on branches please refer to [branches](../../tables/branching).
+For more information on branches please refer to [branches](../branching).
```
Comment (nastra) [inline docs/flink-writes.md:196]: shouldn't this be `../tables/branching`? Same for the spark-writes.md

Comment (manuzhang) [reply docs/flink-writes.md:196]: I'm matching this URL https://iceberg.apache.org/docs/latest/branching/

Comment (nastra) [reply docs/flink-writes.md:196]: I think you actually need to match https://github.com/apache/iceberg-docs/blob/9c5cc5b241c177f153e5e2006eff1b8597c0c097/docs/content/branching-and-tagging.md?plain=1#L5

**Comment by manuzhang** [comment]
@nastra updated. please review again.

**Review by nastra — APPROVED** [review]

**Pull request merged**